### PR TITLE
Implement event loop to reduce number of goroutines in bridge device

### PR DESCRIPTION
**Pull request by jkralik**
## Summary by CodeRabbit

- **New Features**
	- Enhanced device and resource initialization to support event-driven actions.
	- Integrated an event loop mechanism across various components for improved event handling and resource management.
	- Added support for TLS version 1.2 in CoAP gateway tests for enhanced security.

- **Refactor**
	- Unified event handling by removing redundant code and leveraging the new event loop infrastructure.
	- Streamlined resource creation and management to utilize the event loop effectively.
	- Updated logic in various components to use event loop handlers for triggering actions.

- **Tests**
	- Introduced tests for the event loop functionality, ensuring thread safety and correct event processing.
	- Updated existing tests to accommodate the new event loop parameter in resource creation and management functions.

### CODE REVIEW

File: bridge/device/cloud/manager.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ import (
 	"crypto/tls"
 	"crypto/x509"
 	"fmt"
+	"reflect"
 	"sync"
 	"time"
```
[the remark this answers is not in the document]
Comment (jkralik) [reply bridge/device/cloud/manager.go:29]: set to tls.VersionTLS12

File: bridge/service/service.go [hunk 2]
```diff
@@ -90,7 +92,7 @@ func (c *Service) handleDiscoverAllLinks(req *net.Request) (*pool.Message, error
 			}
 		}
 		return links
-	})
+	}, nil)
```
[the remark this answers is not in the document]
Comment (jkralik) [reply bridge/service/service.go]: refactored - thx

File: pkg/eventloop/eventloop_internal_test.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+import (
+	"reflect"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestLoopRun(t *testing.T) {
```
Comment (Danielius1922) [inline pkg/eventloop/eventloop_internal_test.go:11]: I would also add a test with multiple go routines, to test thread-safety

Comment (Danielius1922) [reply pkg/eventloop/eventloop_internal_test.go:11]: @coderabbitai Generate unit tests for eventloop that tests thread-safety

Comment (Danielius1922) [reply pkg/eventloop/eventloop_internal_test.go:11]: @coderabbitai The Remove method should be thread-safe as well, add tests for it too

**Review by Danielius1922 — APPROVED** [review]

File: pkg/eventloop/eventloop_internal_test.go (hunk 3 above)
Comment (jkralik) [reply pkg/eventloop/eventloop_internal_test.go:12]: I added tests but your suggestion doesn't match the code of the implemented event loop.

File: pkg/eventloop/eventloop_internal_test.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+				wg.Done()
+			}))
+			close(ch)
+		}(i)
+	}
+
+	wg.Wait() // Wait for all goroutines to finish
+
+	// Verify that the event loop processed all tasks
+	require.Equal(t, 0, len(*loop.handlers.Load()), "All tasks should be processed")
```
[the remark this answers is not in the document]
Comment (jkralik) [reply pkg/eventloop/eventloop_internal_test.go]: fixed

File: bridge/resources/resource.go [hunk 5]
```diff
@@ -235,6 +198,9 @@ func calcCRC64(body io.ReadSeeker) uint64 {
 }
 
 func (r *Resource) observerHandler(req *net.Request, createSubscription bool) (*pool.Message, error) {
+	if r.loop == nil {
+		return CreateResponseBadRequest(req.Context(), fmt.Errorf("eventloop is nil"))
+	}
```
[the remark this answers is not in the document]
Comment (jkralik) [reply bridge/resources/resource.go]: changed

**Pull request merged**